components/Tile.js

```diff
@@ -0,0 +1,264 @@
+  }
+
+  handleKeyDown = evt => {
+    if (evt.which === 13 || evt.which === 32) {
+      this.setState({
+        clicked: !this.state.clicked
+      });
+    }
+    this.props.handleKeyDown();
+  }
+
+  render() {
+    const { children, href, className, ...other } = this.props;
+    const { clicked } = this.state;
+
+    const classes = classNames('bx--tile', 'bx--tile--clickable', {
+      'bx--tile--is-clicked': clicked
+    }, className);
+
+    return (
+      <a href={href} className={classes} {...other} onClick={this.handleClick} onKeyDown={this.handleKeyDown}>
+        {children}
+      </a>
+    );
+  }
+}
+
+class SelectableTile extends Component {
+  state = {
+    selected: this.props.selected
+  }
+
+  static propTypes = {
+    children: PropTypes.node,
+    className: PropTypes.string,
+    selected: PropTypes.bool,
+    id: PropTypes.string,
+    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
+    name: PropTypes.string,
+    title: PropTypes.string,
+  };
+
+  static defaultProps = {
+    value: 'value',
+    title: 'title',
+    selected: false,
+    handleClick: () => {},
+    handleKeyDown: () => {}
+  };
+
+  handleClick = evt => {
+    const isInput = evt.target === this.input;
+    if (!isInput) {
+      this.setState({
+        selected: !this.state.selected
+      });
+    }
+    this.props.handleClick();
+  }
+
+  handleKeyDown = evt => {
+    if (evt.which === 13 || evt.which === 32) {
+      this.setState({
+        selected: !this.state.selected
+      });
+    }
+    this.props.handleKeyDown();
+  }
+
+  render() {
+    const { children, id, tabIndex, value, name, title, className, ...other } = this.props;
+
+    const classes = classNames('bx--tile', 'bx--tile--selectable', {
+      'bx--tile--is-selected': this.state.selected
+    }, className);
+
+    return (
+      <label htmlFor={id} className={classes} tabIndex={tabIndex} {...other} onClick={this.handleClick} onKeyDown={this.handleKeyDown}>
+        <input ref={input => { this.input = input; }} tabIndex={-1} id={id} className="bx--tile-input" value={value} type="checkbox" name={name} title={title} checked={this.state.selected} />
+        <div className="bx--tile__checkmark">
+          <Icon name="checkmark--glyph" description="Tile checkmark" />
+        </div>
+        <div className="bx--tile-content">
+          {children}
+        </div>
+      </label>
+    );
+  }
+}
+
+class ExpandableTile extends Component {
+  state = {
+    expanded: this.props.expanded,
+    tileMaxHeight: this.props.tileMaxHeight
+  }
+
+  static propTypes = {
+    children: PropTypes.node,
+    className: PropTypes.string,
+    expanded: PropTypes.bool,
+    tabIndex: PropTypes.number
+  };
+
+  static defaultProps = {
+    tabIndex: 0,
+    expanded: false,
+    tileMaxHeight: '0',
+    handleClick: () => {},
+  }
+
+  componentDidMount = () => {
+    this.aboveTheFold = ReactDOM.findDOMNode(this.refs[0]); // eslint-disable-line
```

Comment:
use a `ref` here instead of findDOMNode 

you'll also need to wrap any `ref` usage in an `if`